### DISCUSSION TITLE
Which "special" Cloudflared Options are you using?

### DISCUSSION THREAD

**Discussion by brenner-tobias** [homeassistant-apps/app-cloudflared#191] [Polls]
We want to reduce the complexity of this add-on and acknowledge the new possibilities for Managed Tunnels in the Cloudflare Dashboard.
Therefore, we want to validate the necessity of the following options:

1. _warp_enable_ and _warp_routes_: Warp is generally a rare use-case in the HA context and can also be set-up using a Managed Tunnel from the Cloudflare Dashboard.
2. _custom_config_ and _data_folder_: A custom config gives you the freedom of manually setting everything possible in Cloudflared itself, which is great. Nevertheless, all these options are now also available using a Managed Tunnel from the Cloudflare Dashboard, which makes it easier to use and set-up.
3. _catch_all_service_: A custom catch_all_service can be defined (e.g. an external reverse proxy). This can be very helpful but at the same time can again be archived using a Managed Tunnel from the Cloudflare Dashboard.

So to better understand, what our Users are using, please indicate here, what you are using. If you are using multiple, please vote for the most important one from your perspective.

**Comment by iosoft** [comment]
OPTION: _custom_config and data_folder_
is very important for me. Pls do not remove it.

**Comment by xandervanerven** [comment]
Maybe you can hide these extra options under an "advanced options toggle", so the average user won't be confronted with the options they don't use.

**Comment by lsgv** [comment]
I use warp-enable and routes together with warp on my mobile devices to establish what is in fact a permanent VPN-like connection to my home network when away. I could also do it with a "real" VPN like WireGuard but that would be another add-on and configuration set. I also use warp on my vacation home PC with warp-routes to have permanent connection to my home network. I understand I could also do it with another VPN but the fact is that your add-on serves multiple purposes and I believe that is a good thing. Thanks.

**Reply by ch4d1** [reply]
Same, would be sad to see this feature go.

**Comment by carloszacu** [comment]
it would be nice to be able to share other services in the same server through different port.

in my case I have 5 services including HA in the same server and it would be cool to unify everything in one solution...

awesome job anyway! :)

**Reply by brenner-tobias** [reply]
Thanks for your suggestions. I am not sure though, if I get your point. There are two possibilities here:

1. You want to share multiple services using one URL (e.g. home.mydomain.com) and separate them by different ports (e.g. HA is home.mydomain.com:8123, your Diskstation is home.mydomain.com:5001). This is unfortunately not possible with Cloudflare tunnel. The whole idea of the Cloudflare Tunnel is to use the Cloudflare Reverse proxy to offer wevices using https via. 443. However, you can differentiate using different paths in your domain (e.g. HA is https://home.mydomain.com/ha and your Diskstation is https://home.mydomain.com/ds). You can read about the path ingress rule [here]((e.g. HA is home.mydomain.com:8123, your Diskstation is home.mydomain.com:5001)). In order to use it with the add-on, I suggest using a [Managed Tunnel from the cloudflare dashboard](https://developers.cloudflare.com/cloudflare-one/connections/connect-apps/install-and-setup/tunnel-guide/remote/).
2. The other thing you might want to do is use the tunnel for multiple services in the same server, differentiated by the port. (e.g. ha running on localhost:8123, some website running on localhost:443, PI-Hole Dashboard running on localhost:534. You can make them all available via the Cloudflare tunnel by creating additional_host entries for them or setting them up in the Cloudflare Dashboard (which I recomment). So for example this would look something like this for the bare ingress-rules:
`ingress:
  - hostname: ha.mydomain.com
    service: http://localhost:8123
  - hostname: pi.mydomain.com
    service: http://localhost:534
  - hostname: mydomain.com
    service: http://localhost:443

**Comment by johnshew** [comment]
Great addon.  Nice to have cloudflared connected through HA VM instead of separate installs on my Pi.

Exclusively using the Cloudflare managed options.  

Note that I also use a Cloudflare issued origin certificate and key to enable HTTPS on HA.  That way all web traffic to/from HA is encrypted - including when accessed via VPN or local lan.  To do this I used the managed option to turn off TLS verification.

**Comment by skynet01** [comment]
I think people would use these other services more if they were explained better. The documentation on Cloudflare is not geared toward an everyday user. If it was written in simple man's terms like. "Hey if you want to connect to an ssh server within your network you just need to turn on this thing and set this thing to that" it would be great. 

So I guess if it was presented more as a solution with a few guides rather than a tool you must figure out a lot of these features would be used more.. (either through this add-on or cloud flare portal)  .. which I know is not your job to write those guides :) .. or if those guides are already out there and we just need to link to them

**Reply by DrSpaldo** [reply]
> I think people would use these other services more if they were explained better. The documentation on Cloudflare is not geared toward an everyday user. If it was written in simple man's terms like. "Hey if you want to connect to an ssh server within your network you just need to turn on this thing and set this thing to that" it would be great.
> 
> So I guess if it was presented more as a solution with a few guides rather than a tool you must figure out a lot of these features would be used more.. (either through this add-on or cloud flare portal) .. which I know is not your job to write those guides :) .. or if those guides are already out there and we just need to link to them

Totally agree. I love this addon and it makes it easy as you say. 

The next phase for me is to host another server/service via the hostname option within the addon, the problem is that putting a user/pass or password on top of it isn't as easy as you would hope. I am still trying to figure it out! Hopefully it will be documented or built into the addon for people with less Cloudflare experience :)

**Comment by brenner-tobias** [comment]
Thanks everyone for voting. Looking at the results, I propose the following:

In the next release, we should
1. Suggest everyone to move to Cloudflare Managed Tunnels if they are using warp or a custom_config with a link to a respecting documentation.
2. Marke warp_enable and warp_routes as deprecated with the information that they will be removed within 4 weeks

In one of the following releases, where we remove warp_enable and warp_routes, we will do the same for custom_config:
1. Mark custom_config as deprecated with the information that it will be removed within 4 weeks

The reason for that is the following: the Warp implementation is rather complex and still limited compared to a fully flexible config. Therefor, it is not feasible to support it, since only experienced users will use it that can also be expected to use a Cloudflared Managed tunnel.
The same goes for the custom_config. There really is no benefit to use a custom_config in the add_on other than the DNS entries being created automatically, which is a 1 Minute job. Compared to that, it introduces a lot of complexity and maintenance effort and is definitely way more complicated than from the Cloudflare Dashboard.
On top of that, there is a "one-click" migration function in Cloudflared Dashboard to migrate from a local to a managed tunnel, so everyone can be expected to do that.

Please let me know your thoughts on this.

**Reply by lsgv** [reply]
I believe you're making the right decision(s). There is no point in duplicating warp and custom_config functionality as they can be easily managed from Cloudflare Access and keeping the add-on as simple as possible is a good thing for quality and stability.
With the recent add-on version I've already moved my warp config to Access and that was less than 1 minute work.
Well done and thanks again for your work.

**Reply by pjanning** [reply]
What is the best solution now, if i used Cloudflared and Warp?

**Reply by AlecRust** [reply]
@pjanning if you're using Warp, you want to manage the tunnel within Cloudflare, rather than within this add-on. See details:

https://github.com/brenner-tobias/addon-cloudflared/blob/main/cloudflared/DOCS.md#remote-tunnel-add-on-setup-advanced-setups-only

**Reply by lsgv** [reply]
To manage your tunnel go to cloudflare.com, zero trust (used to be Access) and select Access/Tunnels/Configure. In the Configure tabs Overview and Public Hostname are managed by your HA add-on. You can use the Private Network tab to define which private networks the Tunnel will forward traffic to. For WARP you need to installa the WARP client on your device (computer, iOS, other) and register to your Tunnel account. For WARP to work on your device you need to install a Cloudflare certificate which is explained in the documentation.
<img width="2286" alt="image" src="https://user-images.githubusercontent.com/32908427/198327847-b6cfd497-5f4a-40cf-9eb7-93a451f761c4.png">
